File: finance-client/src/app/api/materialManage.service.ts
import { Injectable } from '@angular/core';
import * as global from '../global';
import {GetResponseService} from './getResponse.service';
import {Observable} from 'rxjs/Observable';
const url = global.baseUrl + '/MaterialManageController';
// 物料匹配关系管理页面
@Injectable()
export class MaterialManageService {
  constructor (private response: GetResponseService) {}
  // 获取未匹配上的物料
  getUnmatchedMaterial(param): Observable<any> {
    return this.response.appPost(param, url + '/getUnmatchedMaterial');
  }
  // 获取物料匹配关系表的数据
  getMatchedMaterial(param): Observable<any> {
    return this.response.appPost(param, url + '/getMatchedMaterial');
  }
  // 添加物料匹配关系
  addMatchedMaterial(param): Observable<any> {
    return this.response.appPost(param, url + '/addMatchedMaterial');
  }
  // 更新物料匹配关系
  updateMatchedMaterial(param): Observable<any> {
    return this.response.appPost(param, url + '/updateMatchedMaterial');
  }
  // 删除物料匹配关系
  deleteMatchedMaterial(param): Observable<any> {
    return this.response.appPost(param, url + '/deleteMatchMaterial');
  }
}

File: finance-server/hengyi-api/src/main/java/com/hengyi/service/MaterialManageService.java
package com.hengyi.service;

import com.hengyi.domain.MaterialMatchDomain;
import com.hengyi.util.QueryDtoBase;

import java.util.List;

/**
 * @author liuyuan
 * @create 2018-05-17 9:35
 * @description 物料管理界面的服务
 **/
public interface MaterialManageService {
    /***
     * 获取未匹配到物料记录数，从而便于进行分页
     * @return
     */
    Long getUnmatchedMaterialCount();
    /***
     * 获取未匹配到的物料信息
     * @return
     */
    List<MaterialMatchDomain> getUnmatchedMaterial(QueryDtoBase queryDtoBase);

    /***
     * 获取已经匹配上的物料匹配关系记录数
     */
    Long getMatchedMaterialCount();

    /***
     * 获取已经有的匹配关系
     * @return
     */
    List<MaterialMatchDomain> getMatchedMaterial(QueryDtoBase queryDtoBase);

    /***
     * 添加物料匹配关系
     * @param materialMatchDomain
     */
    void addMatchedMaterial(MaterialMatchDomain materialMatchDomain);

    /***
     * 修改物料匹配关系
     * @param materialMatchDomain
     */
    void updateMatchedMaterial(MaterialMatchDomain materialMatchDomain);

    /***
     * 删除物料匹配关系
     * @param materialMatchDomain
     */
    void deleteMatchedMaterial(MaterialMatchDomain materialMatchDomain);
}
